consensus/bor/bor.go

```diff
@@ -774,23 +783,62 @@ func (c *Bor) Seal(chain consensus.ChainReader, block *types.Block, results chan
 
 	// Wait until sealing is terminated or delay timeout.
 	log.Trace("Waiting for slot to sign and propagate", "delay", common.PrettyDuration(delay))
+	shouldSeal := make(chan bool)
+	go c.WaitForSealingOp(stop, shouldSeal, number, delay)
 	go func() {
-		select {
-		case <-stop:
+		switch <-shouldSeal {
```

Comment:
Should I `close(shouldSeal)` the channel immediately after this?

Reply:
if it is not going to be used anywhere else then yes, else defer would be a good idea.

Reply:
Done.